### PR TITLE
[chip-test] Test rv_core_ibex_rnd in entropy auto

### PR DESCRIPTION
**Pull request by moidx**
Update `rv_core_ibex_rnd_test()` to use auto_mode entropy complex configuration to reflect the production use-case.

### CODE REVIEW

**Comment by moidx** [comment]
The changes in this PR cause rnd valid bit to timeout. The main change is to configure entropy complex in `auto_mode`.

```console
[2023-04-14T00:56:51Z INFO  opentitanlib::io::uart] set_flow_control to false
I00000 test_rom.c:155] Version: earlgrey_silver_release_v5-10458-g8c3ad7e98, Build Date: 2023-04-13 17:35:58
I00001 test_rom.c:179] TestROM:e96935a5
I00002 test_rom.c:196] Boot strap requested
I00000 test_rom.c:155] Version: earlgrey_silver_release_v5-10458-g8c3ad7e98, Build Date: 2023-04-13 17:35:58
I00001 test_rom.c:179] TestROM:e96935a5
I00002 test_rom.c:236] Test ROM complete, jumping to flash (addr: 20002380)!
I00000 ottf_main.c:110] Running sw/device/tests/rv_core_ibex_rnd_test.c
I00001 rv_core_ibex_rnd_test.c:52] [0] = 0x0
I00002 rv_core_ibex_rnd_test.c:52] [1] = 0x65038123
E00003 rv_core_ibex_rnd_test.c:54] CHECK-STATUS-fail: DeadlineExceeded:["RV_",23]
I00004 status.c:34] FAIL!
```

CC @andreaskurth  @cindychip @msfschaffner

File: sw/device/tests/rv_core_ibex_rnd_test.c [hunk 1]
```diff
@@ -18,20 +19,29 @@
...
-#define MAX_STATUS_CHECKS 1024
+enum {
+  kRandomDataReads = 32,
+  // Timeout (in microseconds) when polling random data or the validity of
+  // random data.  Can currently take up to 80 ms on the FPGA and up to 2.5 s
+  // in Verilator (in simulated time; empirically determined at the time
+  // of commit) due to the rather slow rate of emulated entropy.  In DV, the
+  // emulated entropy rate gets constrained by overriding `rng_srate_value_max`
+  // so that the overall test does not exceed the default timeout of 60 minutes.
+  kTimeoutUsec = 2500000,
```
Comment (andreaskurth) [inline sw/device/tests/rv_core_ibex_rnd_test.c:35]: Note that the high timeout value in Verilator is somewhat artificial: it takes ca. 1.25 M Ibex clock cycles before the core can consume entropy after it has enabled Entropy Source's Auto mode. As SW assumes an Ibex clock frequency of 500 kHz when running in Verilator (see `sw/device/lib/arch/device_sim_verilator.c`), this translates to 2.5 s of "simulated time", so 2.5e6 us. I'm not convinced that using 500 kHz as Ibex clock frequency in Verilator is meaningful, though.

**Review by andreaskurth — APPROVED** [review]
LGTM with the updated timeout I've committed. Also passes CI. I'll wait for someone else to approve as well, since I've now co-authored this PR

**Review by timothytrippel — APPROVED** [review]

File: sw/device/tests/rv_core_ibex_rnd_test.c (hunk 1 above)
Comment (andreaskurth) [reply sw/device/tests/rv_core_ibex_rnd_test.c:35]: I've created issue #18327 to track this

**Pull request merged**